Add a section about BOM in the buildpack authors guide (#304)
buildpacks/docs: content/docs/buildpack-author-guide/create-buildpack/make-buildpack-configurable.md

[builder] ---> Downloading and extracting Ruby 2.5.0
```

> Apart from the build logs, we can also store information about the dependencies that our buildpack has provided in the output image for introspection. We will do that in the next section.

Author: We could add a better follow up here? Not sure if this one is very exciting/correctly worded?

Reviewer: Hmm. It definitely isn't super exciting, but the other page lead ins aren't super exciting either...
How about something like:
```suggestion
Next, let's see how buildpacks can store information about the dependencies provided in the output image for introspection. 
```
